deluge/tests/test_alertmanager.py

```diff
@@ -27,6 +56,34 @@ def handler(alert):
         self.am.register_handler('dummy_alert', handler)
         self.assertEqual(self.am.handlers['dummy_alert'], [handler])
 
+    def test_pop_alert(self):
+        def dummy_alert_1_handler(alert):
+            d.addCallback(self.assertEqual, alert.message, '1')
+            d.callback(None)
+
+        self.am.register_handler('DummyAlert1', dummy_alert_1_handler)
+        self.am.session.set_alerts()
+
+        d = Deferred()
+        return d
+
+    def test_pause_not_pop_alert(self):
+        def on_pause(value):
+            self.am.register_handler('DummyAlert1', dummy_alert_1_handler)
+            d.addCallback(check_alert)
+
+        def check_alert(value):
+            event = self.am._event.isSet()
+            self.assertFalse(event)
+
+        def dummy_alert_1_handler(alert):
+            self.fail()
+
+        d = component.pause(['AlertManager'])
+        self.am.session.set_alerts()
```

Comment:
Since component.pause() is asynchronous, wouldn't you need to call set_alerts() in a callback on the deferred? Otherwise you may be calling set_alerts() before the component has actually been paused.

Reply:
@aresch immediately shows up to comment on an updated 4 year old ticket. 😮

Reply:
@aresch this is what happens when one writes code at 2AM.. fixed.
@gazpachoking lol